### DISCUSSION TITLE
The help system / reference made QuickBasic great, this should have one too.

### DISCUSSION THREAD

**Discussion by stuaxo** [ianatha/bababasic#76] [General & Ideas]
I grew up on the basics of this family, eventually getting onto QuickBasic 4.5

The great about QBASIC, and more-so QuickBasic was the help/reference.

It would be good to have a reference that embodied some of the ideas that embodied it, but adjusted to be appropriate for Android.

it was very quick to look up commands (using keyboard shortcuts there), and you could hit F1 to look up something from the code.

In QuickBasic there were examples in the help too.

**Comment by ianatha** [comment]
This is a great idea, and an awesome feature that would make this app magnitudes more useful.

Do you have any examples of other mobile apps offering in-context documentation? 

This is my first Android/mobile application, and I'm still trying to wrap my head around the limited screen space and whatnot.

**Comment by ianatha** [comment]
Furthermore, another concern is that the app is internationalized (very much on purpose, since my father, who's the primary user, doesn't speak English very well).

Even if it is machine-translated, I'd want any documentation feature to also be available in, at least, Greek, aside from English.

**Comment by stuaxo** [comment]
You can set things up so people can contribute translations, this might be something as basic as markdown in github, or a wiki that get converted into something more useful.

Being fast was a good attribute of the QuickBasic help (and the help in TurboPascal/TurboC++ which were competing in this space).

**Comment by ianatha** [comment]
I have a first pass of this feature -- mostly to experiment with the UX.
It's very limited -- it can only lookup for available help for keywords under your cursor.
Hopefully, more to come soon.

For the time being, it's available on the "Open Testing" Play Store track (v1.2 (112)), so you'd have to join the beta via [https://play.google.com/apps/testing/io.atha.quickbasic](https://play.google.com/apps/testing/io.atha.quickbasic)
